sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubClientBuilderTest.java

```diff
@@ -125,6 +127,27 @@ public void testProxyOptionsConfiguration(String proxyConfiguration, boolean exp
         Assertions.assertEquals(expectedClientCreation, clientCreated);
     }
 
+    @Test
+    public void testConnectionWithAzureNameKeyCredential() {
+        String fullyQualifiedNamespace = "sb-name.servicebus.windows.net";
+        String sharedAccessKeyName = "SharedAccessKeyName test-value";
+        String sharedAccessKey = "SharedAccessKey test-value";
+        String eventHubName = "test-event-hub-name";
+        assertNotNull(new EventHubClientBuilder()
+            .credential(fullyQualifiedNamespace, eventHubName,
+                new AzureNamedKeyCredential(sharedAccessKeyName, sharedAccessKey)));
+    }
+
+    @Test
+    public void testConnectionWithAzureSasCredential() {
+        String fullyQualifiedNamespace = "sb-name.servicebus.windows.net";
+        String sharedAccessSignature = "SharedAccessSignature test-value";
+        String eventHubName = "test-event-hub-name";
+        assertNotNull(new EventHubClientBuilder()
+            .credential(fullyQualifiedNamespace, eventHubName,
```

Comment:
I don't think this will ever be null because we always `return this;`

Reply:
Based on your comment, the new revision has been corrected.